core/src/input_reader.rs

```diff
@@ -12,16 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::{Epoch, RandaoMixIndex, Root, ValidatorIndex, ValidatorInfo};
+use crate::{ConsensusState, Epoch, RandaoMixIndex, Root, ValidatorIndex, ValidatorInfo};
 use alloy_primitives::{B256, aliases::B32};
-use beacon_types::{ChainSpec, EthSpec, Unsigned};
+use beacon_types::{Attestation, ChainSpec, EthSpec, Unsigned};
 use safe_arith::{ArithError, SafeArith};
 use sha2::Digest;
 
-mod ssz_state_reader;
-pub use ssz_state_reader::*;
+mod guest_input_reader;
+pub use guest_input_reader::*;
 
-pub trait StateReader {
+pub trait InputReader {
```

Comment:
We could potentially split this into two traits to add a bit more structure:
The StateReader as before and then `pub trait InputReader: StateReader` with the input specific stuff.
Then on the implementation side nothing would change but the definitions might be a bit more structured.